### PR TITLE
Page object model

### PR DESCRIPTION
**Pull request by lopez997**

### CODE REVIEW

File: src/page/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export { MenuContentPage } from './menu-content.page';
+export { AddressStepPage } from './address-step.page';
+export { BankPaymentPage } from './bank-payment.page';
+export { ProductAddedPage } from './product-added-modal.page';
+export { ProductListPage } from './product-list.page';
+export { SummaryContentPage } from './summary-step.page';
+export { SignInContent } from './sign-in-step.page';
```
Comment (holgiosalos) [inline src/page/index.ts]: add the page suffix: SignInPage or SignInStepPage, that implies renaming the class name.

File: test/buy-tshirt.spec.ts [hunk 2]
```diff
@@ -1,38 +1,55 @@
...
 
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
     await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    // product-added-modal.page
+    await productAddedPage.addProduct();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: I guess you got confused, The modal page is the next step, and the productList page is this step :)

File: test/buy-tshirt.spec.ts [hunk 3]
```diff
@@ -1,38 +1,55 @@
...
     await(browser.sleep(3000));
-
-    await $('#center_column > form > p > button > span').click();
+    // address-step.page
+    await addressContentPage.confirmAddress();
     await(browser.sleep(3000));
-
-    await $('#cgv').click();
+    // order-summary.page
+    await orderSummaryPage.checkOrderSummary();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: If you pay attention, the shipping step should have two actions, the first one, accept the terms of service and the second one the proceed to the payment, hence, this step should be on the `shippingStepPage`

File: test/buy-tshirt.spec.ts [hunk 4]
```diff
@@ -1,38 +1,55 @@
...
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+    // payment-step.page
+    await paymentStepPage.goToPaymentStep();
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+    // bank-payment.page
+    await bankContentPage.checkBankPayment();
     await(browser.sleep(3000));
 
     await expect($('#center_column > div > p > strong').getText())
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: Here you should implement the orderSummary step. That page object should return the title, and here in the test class you should make the verification with the expected title. 

The reason for this is that the idea of the POM is to model the business behavior of an app component, however, the expectations or validations should remain in the test classes. That being said, the right pattern here should be to have a method to retrieve the orderConfirmation from the summary and compare that confirmation with the expected message on the test class.

Comment (lopez997) [reply test/buy-tshirt.spec.ts]: Got it!!

File: test/buy-tshirt.spec.ts [hunk 5]
```diff
@@ -51,8 +50,10 @@ describe('Buy a t-shirt', () => {
     // bank-payment.page
     await bankContentPage.checkBankPayment();
     await(browser.sleep(3000));
-
-    await expect($('#center_column > div > p > strong').getText())
+    // order-summary.page
+    await expect(orderSummaryPage.checkOrderSummary())
       .toBe('Your order on My Store is complete.');
+    /*await expect($('#center_column > div > p > strong').getText())
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: remove this commented code

Comment (lopez997) [reply test/buy-tshirt.spec.ts]: Ok... Sorry

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**